### PR TITLE
Added service team review request support

### PR DESCRIPTION
**Pull request by melinath**
Fixed b/295224401

```release-note:none

```

### CODE REVIEW

File: .ci/magician/cmd/request_service_reviewers.go
Comment (ScottSuarez) [inline .ci/magician/cmd/request_service_reviewers.go:102]:
```suggestion
		previousReviewersMap[reviewer.Login] = map[string][]User
```
Make object declaration more specific

File: .ci/magician/cmd/request_service_reviewers.go
Comment (ScottSuarez) [inline .ci/magician/cmd/request_service_reviewers.go:48]: We probably want to have a short circuit exit condition for the terraform team. Changing generator level stuff might make noise for other teams (depending on the label application process).

File: .ci/magician/cmd/request_service_reviewers.go
Comment (ScottSuarez) [inline .ci/magician/cmd/request_service_reviewers.go:88]: An aside..

How are we thinking about service team reviews. 

My time to review tool isn't taking to account this structure.

What if a service team isn't responding to review requests?

File: .ci/magician/cmd/request_service_reviewers.go
Comment (ScottSuarez) [inline .ci/magician/cmd/request_service_reviewers.go:94]:
```suggestion
	reviewerToRequest := []string{}
```

**Review by ScottSuarez — CHANGES_REQUESTED** [review]

File: .ci/magician/cmd/request_service_reviewers.go
Comment (melinath) [reply .ci/magician/cmd/request_service_reviewers.go:102]: `map[string]struct{}` is the standard way of doing a set in Go - it's slightly more efficient than `map[string]bool`. There isn't a user list here.

File: .ci/magician/cmd/request_service_reviewers.go
Comment (melinath) [reply .ci/magician/cmd/request_service_reviewers.go:88]: go/terraform-resource-owner-pr-notifications has details

File: .ci/magician/cmd/request_service_reviewers.go
Comment (melinath) [reply .ci/magician/cmd/request_service_reviewers.go:48]: I don't think we want to exclude core contributors from this; large generator changes will be excluded by the ">3 teams impacted" condition.

File: .ci/magician/cmd/request_service_reviewers.go
Comment (melinath) [reply .ci/magician/cmd/request_service_reviewers.go]: done

File: .ci/magician/cmd/request_service_reviewers.go
Comment (ScottSuarez) [reply .ci/magician/cmd/request_service_reviewers.go:102]: Apologies for the confusion.

My concern here is that we are initializing `previousReviewersMap[reviewer.Login]` to `struct{}{}`. For readability I would prefer to initialize it to the actually type we intend to use. That way when a new contributor is reading they don't have to gain context on the type definitions and it takes up the same space.

I won't block on this.

**Review by ScottSuarez — APPROVED** [review]

**Comment by ScottSuarez** [comment]
Small notice, unit tests are failing

File: .ci/magician/cmd/request_service_reviewers.go
Comment (melinath) [reply .ci/magician/cmd/request_service_reviewers.go:102]: previousReviewersMap is a `map[string]struct{}` - this line is setting the key `reviewer.Login` (a string) to `struct{}{}` (an empty object of type `struct`), which is the standard way of initializing a set in Go - see https://www.reddit.com/r/golang/comments/xfc33s/when_will_go_get_sets/ for some discussion.

`struct{}{}` is the intended value here; it's never used, it's just a placeholder to indicate that the reviewer is in the set.

I wouldn't want to change this to a `map[string][]User` because we're not using the value, so it doesn't need to hold meaningful data.

I'll rename these variables to `Set` instead of `Map` for clarity.

Please let me know if I'm missing / misunderstanding anything.

Comment (ScottSuarez) [reply .ci/magician/cmd/request_service_reviewers.go]: Ah, you're totally right here. My misunderstanding. I was conflating this with something else in my head. 

Thanks for the clarifications!

**Pull request merged**
